.travis.yml

```diff
@@ -1,40 +1,127 @@
+conditions: v1
+
 language: python
+python:
+- "2.6"
+- "2.7"
+- "3.4"
+- "3.5"
+- "3.6"
+- "pypy2.7-5.10.0"
+- "pypy3.5-5.10.1"
 
 sudo: false
 
-cache: pip
+cache:
+  pip: true
+  directories:
+  - $HOME/.cache/pre-commit
+  - $HOME/.pre-commit
+  - $HOME/Library/Caches/Homebrew
 
-matrix:
-  include:
-    - python: 2.6
-      env: TOXENV=py26
-    - python: 2.7
-      env: TOXENV=py27
-    - python: 3.4
-      env: TOXENV=py34
-    - python: 3.5
-      env: TOXENV=py35
-    - python: 3.6
-      env: TOXENV=py36
-    - python: 3.7-dev
-      env: TOXENV=py37
-    - python: pypy
-      env: TOXENV=pypy
+install:
+- set -e
+- python -m pip.__main__ install cython tox
 
 # build libyaml
+- pushd /tmp
+- git clone https://github.com/yaml/libyaml.git -b 0.2.2-pre1 libyaml
+- cd libyaml
+- ./bootstrap
+- ./configure
+- make
+- make test-all
+- sudo make install
+- popd
+- set +e
+
 before_script:
-- >-
-  cd /tmp
-  && git clone https://github.com/yaml/libyaml.git -b 0.2.2-pre1 libyaml
-  && cd libyaml
-  && ./bootstrap
-  && ./configure
-  && make
-  && make test-all
-  && sudo make install
-  && sudo ldconfig
-  && cd "$TRAVIS_BUILD_DIR"
-
-install: pip install cython tox
-
-script: tox
+- sudo ldconfig
+
+script:
+- python -m tox.__main__
+
+.mixtures:
+- &osx_python
+  if: type IN (api, cron) OR tag IS present
+  os: osx
+  osx_image: xcode9.4
+  language: generic
+  before_install:
+  - brew install zlib readline
+  - brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/cac0ae5/Formula/pyenv.rb || brew upgrade pyenv
+  - &ensure_pyenv_preloaded eval "$(pyenv init -)"
+  - &select_specific_python |
+      if [[ ! "$TRAVIS_PYTHON_VERSION" =~ "dev" ]]
+      then
+        export TRAVIS_PYTHON_VERSION=$(\
+          pyenv install --list | \
+          grep -E '\s\s'"$TRAVIS_PYTHON_VERSION" | grep -vE 'dev|rc' | \
+          tail -n 1 | tr -d '[:space:]'\
+        )
+      fi
+  - &install_python pyenv install --skip-existing --keep --verbose "$TRAVIS_PYTHON_VERSION"
+  - &switch_python pyenv shell "$TRAVIS_PYTHON_VERSION"
+  - &python_version python --version
+  before_script: []
+  before_cache:
+  - brew --cache
+
+jobs:
+  fast_finish: true
+  allow_failures:
+  - os: osx
+    env:
+      TRAVIS_PYTHON_VERSION: "2.6"
+  - os: osx
+    env:
+      TRAVIS_PYTHON_VERSION: "2.7"
+  - os: osx
+    env:
+      TRAVIS_PYTHON_VERSION: "pypy3.5-5.10.1"
+  include:
+  - python: "3.7"
+    dist: xenial
+    sudo: required
+  - <<: *osx_python
+    env:
+      TRAVIS_PYTHON_VERSION: "2.6"
+    python: "2.6"
+  - <<: *osx_python
+    env:
+      TRAVIS_PYTHON_VERSION: "2.7"
+    python: "2.7"
+  - <<: *osx_python
+    env:
+      TRAVIS_PYTHON_VERSION: "3.4"
+    python: "3.4"
+  - <<: *osx_python
+    env:
+      TRAVIS_PYTHON_VERSION: "3.5"
+    python: "3.5"
+  - <<: *osx_python
+    env:
+      TRAVIS_PYTHON_VERSION: "3.6"
+    python: "3.6"
+  - <<: *osx_python
+    env:
+      TRAVIS_PYTHON_VERSION: "3.7.0"
```

Comment:
Current on Travis CI is Python 3.7.1...  Perhaps is it best to just remove the "micro" version number.